addons/l10n_id/models/qris_transaction.py

```diff
@@ -0,0 +1,145 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import pytz
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+# Using different field for reference for different models
+MODEL_ID_FIELDS = {
+    'account.move': 'id',
+    'pos.order': 'uuid'
+}
+
+MODEL_COMPLETED_CHECK = {
+    'account.move': lambda record: record.payment_state == record._get_invoice_in_payment_state(),
+    'pos.order': lambda record: record.state in ('paid', 'done', 'invoiced')
+}
+class QRISTransaction(models.Model):
+    """QRIS Transaction
+
+    General table to store a certian unique transaction with QRIS details attached
+    """
+    _name = "l10n_id.qris.transaction"
+    _description = "Record of QRIS transactions"
+
+    # trx_uuid = fields.Char()
+    model = fields.Char()  # payment in respond to which model
+    model_id = fields.Char()  # id/uuid
+
+    qris_invoice_details = fields.Json(
+        string="QRIS Transaction Details",
+        readonly=True,
+        export_string_translation=False,
+    )
```

Comment:
This shouldn't be needed anymore.
One move/order will have one qris transaction for each invoice details it had before.
So instead of storing as json, we store as a new object